x-pack/plugins/ml/public/application/explorer/anomalies_map.tsx

```diff
@@ -0,0 +1,252 @@
+      },
+      isTimeAware: true,
+    },
+    visible,
+    type: 'VECTOR',
+  };
+};
+
+interface Props {
+  anomalies: AnomaliesTableRecord[];
+  jobIds: string[];
+}
+
+interface MLEMSTermJoinConfig extends EMSTermJoinConfig {
+  jobId: string;
+}
+
+export const AnomaliesMap: FC<Props> = ({ anomalies, jobIds }) => {
+  const [EMSSuggestions, setEMSSuggestions] = useState<
+    Array<MLEMSTermJoinConfig | null> | undefined
+  >();
+  const {
+    services: { maps: mapsPlugin },
+  } = useMlKibana();
+
+  const getEMSTermSuggestions = useCallback(async (): Promise<void> => {
+    if (!mapsPlugin) return;
+
+    const suggestions = await Promise.all(
+      jobIds.map(async (jobId) => {
+        const entityValues = new Set<string>();
+        let entityName;
+        for (let i = 0; i < anomalies.length; i++) {
+          if (
+            jobId === anomalies[i].jobId &&
+            anomalies[i].entityValue !== '' &&
+            anomalies[i].entityValue !== undefined &&
+            anomalies[i].entityName !== '' &&
+            anomalies[i].entityName !== undefined
+          ) {
+            entityValues.add(anomalies[i].entityValue);
+
+            if (!entityName) {
+              entityName = anomalies[i].entityName;
+            }
+          }
+
+          if (
+            // convert to set so it's unique values
+            entityValues.size === MAX_ENTITY_VALUES
+          )
+            break;
+        }
+
+        const suggestion: EMSTermJoinConfig | null = await mapsPlugin.suggestEMSTermJoinConfig({
+          emsLayerIds: COMMON_EMS_LAYER_IDS,
+          sampleValues: Array.from(entityValues),
+          sampleValuesColumnName: entityName || '',
+        });
+        if (suggestion) {
+          return { jobId, ...suggestion };
+        }
+        return suggestion;
+      })
+    );
+
+    setEMSSuggestions(suggestions.filter((s) => s !== null));
+  }, [...jobIds]);
+
+  useEffect(
+    function getInitialEMSTermSuggestions() {
+      if (anomalies && anomalies.length > 0) {
+        getEMSTermSuggestions();
+      }
+    },
+    [...jobIds]
+  );
+
+  const layerList = useMemo(() => {
+    let layers: VectorLayerDescriptor[] = [];
+    // Loop through suggestions list and make a layer for each
+    if (EMSSuggestions?.length) {
+      let count = 0;
+      layers = EMSSuggestions.reduce(function (result, suggestion) {
+        if (suggestion) {
+          const visible = count === 0;
+          result.push(getChoroplethAnomaliesLayer(anomalies, suggestion, visible));
+          count++;
+        }
+        return result;
+      }, [] as VectorLayerDescriptor[]);
+    }
+    return layers;
+  }, [EMSSuggestions, anomalies.length]);
+
+  if (EMSSuggestions?.length === 0) {
+    return null;
+  }
+
+  return (
+    <>
+      <EuiPanel>
+        <EuiAccordion
+          id="mlAnomalyExplorerAnomaliesMapAccordionId"
+          initialIsOpen={true}
+          buttonContent={
+            <EuiTitle className="panel-title">
```

Comment:
Would be useful to add an info tooltip here, for example to explain that the map colors indicate count of anomalies.

Reply:
Added tooltip in a21b1ce735b46c25818b38d0a1c497004e261d57